src/coreclr/nativeaot/Runtime/i386/ExceptionHandling.asm

```diff
@@ -100,6 +100,10 @@ FASTCALL_ENDFUNC
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 FASTCALL_FUNC  RhpThrowEx, 4
 
+        mov         edi, 1                  ;; edi = ExKind.Throw
```

Comment:
I do not think you can use edi register here. `edi` register is callee saved register. The original value needs to be preserved in the PAL_LIMITED_CONTEXT. This can use edx. It means that you will need to move `mov         edx, [esp+4]` lower and change it use something else, say `eax`.

Reply:
Yep